### PR TITLE
[develop] Carlos - Decoupling code

### LINKED ISSUE #40: Decoupling Context from EndPointEnum
- [x] Decoupling context
- [x] Solve problems in all dependent classes

### PR DESCRIPTION
**Pull request by carlosmc23**

### CODE REVIEW

**Review by Snip-77 — COMMENTED** [review]
Seems good to me but have to test this code first

File: src/test/java/org/fundacionjala/trello/runner/Runner.java [hunk 1]
```diff
@@ -39,5 +41,6 @@ public void beforeAllScenarios() {
     @AfterTest
     public void afterAllScenarios() {
         // Code executed after features execution.
+        getDriver().quit();
```
Comment (luqp) [inline src/test/java/org/fundacionjala/trello/runner/Runner.java]: I get a `NullPointerException` error in this line, when I run all tests by the `Runner` Class

**Review by luqp — COMMENTED** [review]
It looks good but I believe that the `WebDriverAction` class should be in the `driver` folder
and `driver` folder should be in the `core` folder

File: src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java [hunk 2]
```diff
@@ -1,5 +1,6 @@
 package org.fundacionjala.trello.hooks;
 
+
```
Comment (Snip-77) [inline src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java]: Unnecessary space here

File: src/test/java/org/fundacionjala/trello/runner/Runner.java [hunk 3]
```diff
@@ -39,5 +41,6 @@ public void beforeAllScenarios() {
     @AfterTest
     public void afterAllScenarios() {
```
Comment (Snip-77) [inline src/test/java/org/fundacionjala/trello/runner/Runner.java]: Getting a null pointer exception here when running tests

File: src/test/java/org/fundacionjala/trello/stepdefs/BoardsStepDef.java [hunk 4]
```diff
@@ -1,22 +1,20 @@
...
 import io.cucumber.java.en.When;
-import org.fundacionjala.trello.context.Context;
 import org.fundacionjala.trello.driver.SharedDriver;
 import org.fundacionjala.trello.pages.home.BoardsPage;
 import org.fundacionjala.trello.pages.home.sections.Section;
 import org.fundacionjala.trello.pages.home.sections.TeamSection;
 
 import static org.fundacionjala.trello.driver.DriverFactory.getDriver;
 
+
```
Comment (Snip-77) [inline src/test/java/org/fundacionjala/trello/stepdefs/BoardsStepDef.java]: Unnesesary space here

File: src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java [hunk 5]
```diff
@@ -36,7 +31,7 @@ public CardStepDefs(final SharedDriver sharedDriver, final Context context) {
      * Creates a card with specific data.
      *
      * @param listName expected name of the list where card will be created.
-     * @param data expected card data.
+     * @param data     expected card data.
```
Comment (Snip-77) [inline src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java:34]: Unnecesarry spaces here

File: src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java [hunk 6]
```diff
@@ -51,7 +46,7 @@ public void iCreateCardWith(final String listName, final Map<String, String> dat
     /**
      * Validates the card creation.
      *
-     * @param listName expected list name where to find the card.
+     * @param listName     expected list name where to find the card.
```
Comment (Snip-77) [inline src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java:49]: Unnecesarry spaces here

File: src/test/resources/features/cards/createCard.feature [hunk 7]
```diff
@@ -14,7 +14,9 @@ Feature: Card
     Given I log in with my Trello account as "user1"
     And I navigate to boards menu from header
     And I open the "New board" board
+#    When I create a Card on "Jacoco" list
```
Comment (Snip-77) [inline src/test/resources/features/cards/createCard.feature:17]: Lets errase this comentaries.

**Review by Snip-77 — COMMENTED** [review]
Have some minor changes, review the commentaries please

File: src/test/java/org/fundacionjala/trello/runner/Runner.java (hunk 1 above)
Comment (carlosmc23) [reply src/test/java/org/fundacionjala/trello/runner/Runner.java]: yes, you right.. Its solve now

File: src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java (hunk 2 above)
Comment (carlosmc23) [reply src/test/java/org/fundacionjala/trello/hooks/CommonHooks.java]: done! thanks

File: src/test/java/org/fundacionjala/trello/runner/Runner.java (hunk 3 above)
Comment (carlosmc23) [reply src/test/java/org/fundacionjala/trello/runner/Runner.java]: yes its solve thanks

File: src/test/java/org/fundacionjala/trello/stepdefs/BoardsStepDef.java (hunk 4 above)
Comment (carlosmc23) [reply src/test/java/org/fundacionjala/trello/stepdefs/BoardsStepDef.java]: Done!! thanks

File: src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java (hunk 5 above)
Comment (carlosmc23) [reply src/test/java/org/fundacionjala/trello/stepdefs/CardStepDefs.java:34]: This is the configuration forma of check stile

File: src/test/resources/features/cards/createCard.feature (hunk 7 above)
Comment (carlosmc23) [reply src/test/resources/features/cards/createCard.feature:17]: I think you need to apply this changes, in order to solve our trainer observations.

**Review by Snip-77 — APPROVED** [review]

**Pull request merged**